src/test/rgw/rgw_multi/tests.py

```diff
@@ -387,6 +395,25 @@ def set_master_zone(zone):
     log.info('Set master zone=%s, waiting %ds for reconfiguration..', zone.name, config.reconfigure_delay)
     time.sleep(config.reconfigure_delay)
 
+def enable_bucket_sync(zone, bucket_name):
+    cmd = ['bucket', 'sync', 'enable', '--bucket', bucket_name] + zone.zone_args()
+    zone.cluster.admin(cmd)
+
+def disable_bucket_sync(zone, bucket_name):
+    cmd = ['bucket', 'sync', 'disable', '--bucket', bucket_name] + zone.zone_args()
+    zone.cluster.admin(cmd)
+
+def check_buckets_sync_status_obj_not_exist(zone, buckets):
+    a = '|'
+    cmd =  ['log', 'list'] + zone.zone_arg() + ['|grep ', '"' + a.join(buckets) + '"']
+    for _ in range(config.checkpoint_retries):
+        log_list, ret = zone.cluster.admin(cmd, check_retcode=False)
```

Comment:
yeah, teuthology is not handling this part well: [example results](http://qa-proxy.ceph.com/teuthology/cbodley-2017-07-18_09:23:33-rgw-wip-rgw-per-bucket-sync-distro-basic-mira/1412703/teuthology.log)

the [Cluster implementation for teuthology](https://github.com/ceph/ceph/blob/master/qa/tasks/rgw_multisite.py#L186-L192) is adding extra `--cluster` and `--debug-rgw` arguments after the pipe (and since this happens in a loop, those arguments get appended multiple times: `--cluster c1 --debug-rgw 0 --cluster c1 --debug-rgw 0 --cluster c1 --debug-rgw 0 ...`)

as gross as it is, can you remove the `|grep` part, and search the `log_list` string manually? it should also verify that the object name starts with `bucket.sync-status`. in the run above, i see that it matched an object named `2017-07-18-14-c772d205-b341-46d5-af49-43c21450c74f.4196.9-hoxbdk-35` - i'm not sure where that came from

then we can also add `read_only=True` to `zone.cluster.admin()`